roles/common/tasks/kernel-tuning.yml

```diff
@@ -28,6 +28,13 @@
               line="GRUB_CMDLINE_LINUX_DEFAULT=\"\""
   notify: update grub config
 
+- name: "Limit available cores via grub"
+  lineinfile: dest=/etc/default/grub
+              regexp="^GRUB_CMDLINE_LINUX_DEFAULT="
+              line="GRUB_CMDLINE_LINUX_DEFAULT=\"{{ cpulimit }}\""
```

Comment:
This should be `maxcpu={{ cpulimit }}`. You were missing the maxcpu part.

However, this should go into the `GRUB_CMDLINE_LINUX` line, not this one.